deploy/deploy.go

```diff
@@ -215,48 +218,20 @@ type KindSpec struct {
 	Retain                   bool              `yaml:"retain"`
 	Wait                     time.Duration     `yaml:"wait"`
 	Kubecfg                  string            `yaml:"kubecfg"`
-	DeployWithClient         bool              `yaml:"deployWithClient"`
 	GoogleArtifactRegistries []string          `yaml:"googleArtifactRegistries"`
 	ContainerImages          map[string]string `yaml:"containerImages"`
 }
 
 func (k *KindSpec) Deploy(ctx context.Context) error {
-	provider := newProvider()
-	if k.Recycle {
-		clusters, err := provider.List()
-		if err != nil {
-			return err
-		}
-		for _, v := range clusters {
-			if k.Name == v {
-				log.Infof("Recycling existing cluster: %s", v)
-				return nil
-			}
-		}
+	if _, err := execLookPath("kind"); err != nil {
```

Comment:
maybe we should codify for each spec a "depends" on func like you did above but attach it to the struct?

Reply:
yeah I agree, we can have a depends for the spec, additionally options such as googleArtifactRegistry access requires an additional dep (gcloud) which we could check in the depends func based on the optional fields provided

Reply:
done